css/css-conditional/idlharness.html

```diff
@@ -14,18 +14,31 @@
 <body>
   <div id="log"></div>
 
+  <style>
+    @media screen { body { font-size: normal; } }
+    @supports (display: grid) { body { font-size: normal; } }
+  </style>
+
   <script>
     'use strict';
-    promise_test(async () => {
-      const idl = await fetch('/interfaces/css-conditional.idl').then(r => r.text());
-      const cssom = await fetch('/interfaces/cssom.idl').then(r => r.text());
-      const dom = await fetch('/interfaces/dom.idl').then(r => r.text());
-      const idl_array = new IdlArray();
-      idl_array.add_idls(idl);
-      idl_array.add_dependency_idls(cssom);
-      idl_array.add_dependency_idls(dom);
-      idl_array.test();
-    }, 'Test css-conditional IDL implementation');
+
+    idl_test(
+      ['css-conditional'],
+      ['cssom', 'dom'],
+      idl_array => {
+        idl_array.add_objects({
+          CSSRule: ['rule'],
+          CSSMediaRule: ['media'],
+          CSSSupportsRule: ['supports'],
+        });
+        idl_array.prevent_multiple_testing('CSSRule');
+
+        self.media = document.styleSheets[0].cssRules[0];
+        self.supports = document.styleSheets[0].cssRules[1];
+        self.rule = media.cssRules[0];
```

Comment:
Does this actually work, is this object a CSSRule and not some more specific type?